### DISCUSSION TITLE
Making max_results=8 an optional argument for context_compressor

### DISCUSSION THREAD

**Discussion by a-yuksel** [assafelovic/gpt-researcher#782] [Ideas]
Hi everyone,
I noticed that the max_results parameter is set to 8 in the context_compressor function at the following location:

https://github.com/assafelovic/gpt-researcher/blob/e9378e9ddbf4b893a35b791da354a51910e478c5/gpt_researcher/master/agent.py#L398C1-L401C1

Is this setting intentional? Additionally, would it be possible to allow users to supply their own value for this parameter?

Thanks for your help!

**Comment by timothylowjc** [comment]
Hi,

I'm having the same problem as well, might make sense to convert this to an issue
